app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/DisplayImageSetFragment.kt

```diff
@@ -1,112 +1,134 @@
+import androidx.navigation.fragment.navArgs
+import com.github.HumanLearning2021.HumanLearningApp.R
+import com.github.HumanLearning2021.HumanLearningApp.databinding.FragmentDisplayImageSetBinding
+import com.github.HumanLearning2021.HumanLearningApp.databinding.FragmentLearningBinding
+import com.github.HumanLearning2021.HumanLearningApp.hilt.Demo2Database
+import com.github.HumanLearning2021.HumanLearningApp.hilt.DummyDatabase
+import com.github.HumanLearning2021.HumanLearningApp.hilt.ScratchDatabase
+import com.github.HumanLearning2021.HumanLearningApp.model.CategorizedPicture
+import com.github.HumanLearning2021.HumanLearningApp.model.Category
+import com.github.HumanLearning2021.HumanLearningApp.model.DatabaseManagement
+import com.github.HumanLearning2021.HumanLearningApp.view.learning.LearningFragmentArgs
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class DisplayImageSetFragment: Fragment() {
+    private lateinit var parentActivity: FragmentActivity
+
+    @Inject
+    @Demo2Database
+    lateinit var dBManagement: DatabaseManagement
+
+    private var categorizedPicturesList = setOf<CategorizedPicture>()
+    private lateinit var datasetId: String
+    private lateinit var category: Category
+
+    private val args: DisplayImageSetFragmentArgs by navArgs()
+    private var _binding: FragmentDisplayImageSetBinding? = null
+    private val binding get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        parentActivity = requireActivity()
+        _binding = FragmentDisplayImageSetBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        category = args.category
+        datasetId = args.datasetId
+
+        lifecycleScope.launch {
+            categorizedPicturesList = dBManagement.getAllPictures(category)
+            if (categorizedPicturesList.isNotEmpty()) {
+                parentActivity.findViewById<TextView>(R.id.display_image_set_name).text =
+                    (categorizedPicturesList.elementAt(0)).category.name
+
+                val displayImageSetAdapter =
+                    DisplayImageSetAdapter(
+                        categorizedPicturesList,
+                        parentActivity
+                    )
+
+                parentActivity.findViewById<GridView>(R.id.display_image_set_imagesGridView).adapter =
+                    displayImageSetAdapter
+                setPictureItemListener()
+            }
+        }
+
+        val callback = object : OnBackPressedCallback(true){
+            override fun handleOnBackPressed() {
+                findNavController().popBackStack()
+            }
+        }
```

Comment:
lambda?

Reply:
I'm not sure how to convert this to a lambda but feel free to do it.